rule_match.go

```diff
@@ -35,6 +35,10 @@ type MatchData struct {
 	Value string
 }
 
+func (m MatchData) isNil() bool {
```

Comment:
It would be cool to add some documentation on why use isNil

Reply:
That's ok. I'll add it.